internal/clients/vault_secrets_preview.go

```diff
@@ -61,27 +61,34 @@ func OpenVaultSecretsAppSecrets(ctx context.Context, client *Client, loc *shared
 
 	var secrets *secret_service.OpenAppSecretsOK
 	var err error
-	for attempt := 0; attempt < retryCount; attempt++ {
-		secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
-		if err != nil {
-			var serviceErr *secret_service.OpenAppSecretDefault
-			ok := errors.As(err, &serviceErr)
-			if !ok {
+	var result []*secretmodels.Secrets20231128OpenSecret
+
+	for {
+		for attempt := 0; attempt < retryCount; attempt++ {
+			secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
+			if err != nil {
+				var serviceErr *secret_service.OpenAppSecretDefault
+				ok := errors.As(err, &serviceErr)
+				if !ok {
+					return nil, err
+				}
+				if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
+					continue
+				}
 				return nil, err
 			}
-			if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
-				continue
+			if secrets == nil {
+				return nil, errors.New("unable to get secrets")
 			}
-			return nil, err
+			result = append(result, secrets.GetPayload().Secrets...)
+			pagination := secrets.GetPayload().Pagination
+			if pagination == nil || pagination.NextPageToken == "" {
+				return result, nil
+			}
+			params.PaginationNextPageToken = &pagination.NextPageToken
```

Comment:
It seems that the intention here is to read all secrets of an app. If that is the case, can we bump the pageSize parameter of the pagination to its max value to have limited rounds of the API call?

Reply:
If you think it is a good idea, you would also need to change the number secrets generated in your new test

Reply:
Thank you for the suggestion @hghaf099. 
It is a good idea but the optimization is unhelpful if we increase the max size in future and forgot to update the provider.
More details in slack thread here: https://hashicorp.slack.com/archives/C06P685734L/p1728399960978739